src/xtgeo/grid3d/_grdecl_format.py

```diff
@@ -1,12 +1,21 @@
+from __future__ import annotations
+
 import warnings
+from collections.abc import Iterator
 from contextlib import contextmanager
+from typing import TYPE_CHECKING
 
 from xtgeo.common import null_logger
 
+if TYPE_CHECKING:
+    from io import TextIOWrapper
+
+    from xtgeo.common.types import FileLike
+
 logger = null_logger(__name__)
 
 
-def split_line(line):
+def split_line(line: str) -> Iterator[str]:
     """
```

Comment:
I think this is actually a Generator -- which is an iterator -- but a subclass of one.